internal/uvm/create_wcow.go

```diff
@@ -343,8 +340,15 @@ func prepareSecurityConfigDoc(ctx context.Context, uvm *UtilityVM, opts *Options
 		}
 	}
 
-	memoryBacking := hcsschema.MemoryBackingType_PHYSICAL
-	doc.VirtualMachine.ComputeTopology.Memory.Backing = &memoryBacking
+	if opts.IsolationType != "SecureNestedPaging" {
+		// With these options SNP UVM doesn't boot.
+		memoryBacking := hcsschema.MemoryBackingType_PHYSICAL
+		doc.VirtualMachine.ComputeTopology.Memory.Backing = &memoryBacking
+		doc.VirtualMachine.ComputeTopology.Memory.AllowOvercommit = opts.AllowOvercommit
+		// EnableHotHint is not compatible with physical.
+		doc.VirtualMachine.ComputeTopology.Memory.EnableHotHint = opts.AllowOvercommit
```

Comment:
`handleWCOWSecurityPolicy` sets `AllowOvercommit` to false, so we don't need to explicitly set that here.

Do we not need physically backed memory for VBS mode?

Reply:
Maybe I'm missing something, but the change is only for SNP. VBS will keep having memoryBacking with this change.